src/api/spec/features/webui/comment_locks_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'browser_helper'
+
+RSpec.describe 'CommentLocks' do
+  let!(:moderator_user) { create(:moderator) }
+
+  before do
+    Flipper.enable(:content_moderation)
+  end
+
+  context 'on a request' do
+    let(:submitter) { create(:confirmed_user, :with_home, login: 'kugelblitz') }
+    let(:receiver) { create(:confirmed_user, :with_home, login: 'titan') }
+    let(:target_project) { receiver.home_project }
+    let(:target_package) { create(:package, name: 'goal', project_id: target_project.id) }
+    let(:source_project) { submitter.home_project }
+    let(:source_package) { create(:package, name: 'ball', project_id: source_project.id) }
+    let(:bs_request) { create(:delete_bs_request, target_project: target_project, description: 'a long text - ' * 200, creator: submitter) }
+    let(:user) { create(:confirmed_user, :with_home, login: 'burdenski') }
+
+    describe 'with comments enabled' do
+      context 'a moderator user' do
+        before do
+          login moderator_user
+          visit request_show_path(bs_request)
+        end
+
+        it 'checks comments are enabled' do
+          find_button('Lock comments')
+          expect(page).not_to have_text('Commenting on this is locked. You can remove the lock by clicking on the button below.')
+        end
+
+        it 'locks comments' do
+          find_button('Lock comments').click
+          find_button('Lock').click
+          expect(page).to have_text('Commenting on this is locked. You can remove the lock by clicking on the button below.')
+        end
+      end
+
+      context 'a non-moderator user' do
+        before do
+          login user
+          visit request_show_path(bs_request)
+        end
+
+        it 'cannot lock comments' do
+          expect(page).not_to have_button('Lock comments')
+        end
+
+        it 'can comment' do
+          expect(page).not_to have_text('Commenting on this is locked')
+          fill_in 'new_comment_body', with: 'Comment Body'
+          find_button('Add comment')
+        end
+      end
+    end
+
+    describe 'with comments disabled' do
```

Comment:
I think when you say "comments disabled" you mean "comments locked". If that's the case better use just one naming for all so lock/unlock instead of lock/unlock/enable/disable. WDYT?

Reply:
You're right. I was carried away by the fact in the end you cannot comment, but it's still about `comment lock`. Fixing...